src/cache/cloud_fraction.jl

```diff
@@ -139,13 +137,17 @@ NVTX.@annotate function set_cloud_fraction!(
 
     FT = eltype(params)
     thermo_params = CAP.thermodynamics_params(params)
-    (; ᶜts, ᶜmixing_length, cloud_diagnostics_tuple) = p.precomputed
+    (; ᶜts, cloud_diagnostics_tuple) = p.precomputed
     (; turbconv_model) = p.atmos
 
     # TODO - we should make this default when using diagnostic edmf
     # environment
     diagnostic_covariance_coeff = CAP.diagnostic_covariance_coeff(params)
 
+    ᶜmixing_length = p.scratch.ᶜtemp_scalar_6
+    ᶜmixing_length_lazy = mixing_length(Y, p)
+    ᶜmixing_length = Base.Broadcast.materialize(ᶜmixing_length_lazy)
```

Comment:
```suggestion
    ᶜmixing_length .= mixing_length(Y, p)
```
This line will allocate as it is